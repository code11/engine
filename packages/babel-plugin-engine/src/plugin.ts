import * as Babel from "@babel/core";
import { VariableDeclaratorVisitor } from "./visitors";

export type PluginConfig = {
  viewLibrary: string;
};

type pluginType = (babel: typeof Babel, state: PluginConfig) => {};

export const plugin: pluginType = (babel, state) => {
  return {
    name: "babel-plugin-engine",
    visitor: {
      VariableDeclarator: VariableDeclaratorVisitor(babel, state),
    },
  };
};
