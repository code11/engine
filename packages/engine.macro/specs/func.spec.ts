import pluginTester from "babel-plugin-tester";
import prettier from "prettier";
import plugin from "babel-plugin-macros";

const macroFile = "'@c11/engine.macro'";
const config = require("./babelConfig.json");

pluginTester({
  plugin,
  babelOptions: { filename: __filename, ...config },
  formatResult: (result: any) => {
    return prettier.format(result, {
      parser: "babel",
    });
  },
  tests: {
    "should support composed types: functions": {
      code: `
        import { producer } from ${macroFile}
        const result: producer = ({
          a1 = get.foo || '123' && '123'
        }) => { }
      `,
      snapshot: true,
    },
  },
});
