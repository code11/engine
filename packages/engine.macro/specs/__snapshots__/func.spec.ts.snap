// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unknown plugin should support composed types: functions: should support composed types: functions 1`] = `

import { producer } from '@c11/engine.macro'
const result: producer = ({
  a1 = get.foo || '123' && '123'
}) => { }

      ↓ ↓ ↓ ↓ ↓ ↓

"use strict";

var result = {
  args: {
    type: "STRUCT",
    value: {
      a1: {
        type: "FUNC",
        value: {
          params: [
            {
              type: "GET",
              value: "value",
            },
          ],
          fn: "fn",
        },
      },
    },
    meta: {},
  },
  fn: function fn(_ref) {
    var a1 = _ref.a1;
  },
};


`;
